api/specs/web-server/_wallets.py

```diff
@@ -90,6 +93,55 @@ async def cancel_payment(wallet_id: WalletID, payment_id: PaymentID):
     ...
 
 
+### Wallets payment-methods
+
+
+@router.post(
+    "/wallets/{wallet_id}/payments-methods:init",
+    response_model=Envelope[PaymentMethodInit],
+)
+async def init_creation_of_payment_method(wallet_id: WalletID):
+    ...
+
+
+@router.post(
+    "/wallets/{wallet_id}/payments-methods/{payment_method_id}:cancel",
+    status_code=status.HTTP_204_NO_CONTENT,
+    response_description="Successfully cancelled",
+)
+async def cancel_creation_of_payment_method(
+    wallet_id: WalletID, payment_method_id: PaymentMethodID
+):
+    ...
+
+
+@router.get(
+    "/wallets/{wallet_id}/payments-methods",
+    response_model=Envelope[list[PaymentMethodGet]],
+)
+async def list_payments_methods(wallet_id: WalletID):
+    """Lists all payments method associated to `wallet_id`"""
+
+
+@router.get(
+    "/wallets/{wallet_id}/payments-methods/{payment_method_id}",
+    response_model=Envelope[PaymentMethodGet],
+)
+async def get_payment_method(wallet_id: WalletID, payment_method_id: PaymentMethodID):
+    ...
```

Comment:
Q: will there be a use case also for updating?

Reply:
I'd say like everybody does, remove and add again.

Reply:
No. We want to limit functionality to add/delete payment-methods (credit cards) but not changing them